src/test/java/com/mendix/audittrail/tests/TestAuditConstants.java

```diff
@@ -0,0 +1,99 @@
+import java.util.Arrays;
+
+import static org.mockito.Mockito.*;
+
+public class TestAuditConstants extends TestAuditBase {
+	@Test
+	public void testLogUseDecimalScientificNotationIsTrue() throws CoreException {
+		context = constructMockContext(context, true);
+		// Create new record
+		final Company company = createBaseCompany();
+
+		// Assert log was created
+		final ExpectedLog expectedLog = createExpectedLog(TypeOfLog.Add, company).addAttribute(Dec, "1E+3");
+
+		final ActualLog actualLog = ActualLog.getLastLog(context, company.getMendixObject().getId().toLong());
+		expectedLog.verify(actualLog);
+	}
+
+	@Test
+	public void testLogUseDecimalScientificNotationIsFalse() throws CoreException {
+		context = constructMockContext(context, false);
+		// Create new record
+		final Company company = createBaseCompany();
+
+		// Assert log was created
+		final ExpectedLog expectedLog = createExpectedLog(TypeOfLog.Add, company).addAttribute(Dec, "1000");
+
+		final ActualLog actualLog = ActualLog.getLastLog(context, company.getMendixObject().getId().toLong());
+		expectedLog.verify(actualLog);
+	}
+	
+	private IContext constructMockContext(IContext context, Boolean constantValue) {
+		IContext spyContext = spy(context);
+		com.mendix.core.conf.Configuration configSpy = spy(spyContext.getSystemConfiguration());
+		when(spyContext.getSystemConfiguration()).thenReturn(configSpy);
+		when(spyContext.clone()).thenReturn(spyContext);
+		when(configSpy.getConstantValue("AuditTrail.LogUseDecimalScientificNotation")).thenReturn(constantValue);
+		return spyContext;
+	}
+
+	private Company createBaseCompany(final Group... groups) throws CoreException {
+		final Company company = new Company(context);
+		
+		company.setName(NAME);
+		company.setCompanyNr(COMPANY_NR);
+		company.setDec(BigDecimal.valueOf(1000));
+
+		company.setCompany_Group(Arrays.asList(groups));
+
+		company.commit();
+		return company;
+	}
+
+	private ExpectedLog createExpectedLog(final TypeOfLog typeOfLog, final Company company,
+																				final IMendixObject... groupObjects) {
+		if (typeOfLog.equals(TypeOfLog.Add)) {
+			return new ExpectedLog(typeOfLog, Company.entityName, admin, initialDate, Company.entityName)
+					.addAttribute(Name, NAME).addAttribute(CompanyNr, COMPANY_NR)
+					.addAttribute(InternNr, company.getInternNr())
+					.addAttribute(Dec, "1E+3").addAttribute(Number, 0)
+					.addAttribute(Founded, "")
+					.addReferences(Company_Group, context, MemberType.ReferenceSet, groupObjects);
+		} else {
+			return new ExpectedLog(typeOfLog, Company.entityName, admin, initialDate, Company.entityName)
+					.keepAttribute(Name, NAME).keepAttribute(CompanyNr, COMPANY_NR)
+					.keepAttribute(InternNr, company.getInternNr())
+					.keepAttribute(Dec, 0).keepAttribute(Number, 0)
+					.keepAttribute(Founded, "")
+					.keepReferences(Company_Group, context, MemberType.ReferenceSet, groupObjects);
+		}
+	}
+	private static final String NAME = "Company";

```

Comment:
We now have a mixup with constants and literals. Maybe `"1E+3"` and `"1000"` should also become constants?